ray_image_preprocessing.py

```diff
@@ -0,0 +1,136 @@
+"""
+Preprocessing helper functions for the MAPLE pipeline.
+
+Project: Permafrost Discovery Gateway: Mapping Application for Arctic Permafrost Land Environment(MAPLE)
+PI      : Chandi Witharana
+"""
+
+
+import os
+from pathlib import Path
+import shutil
+from typing import Dict, Any
+
+import numpy as np
+from osgeo import gdal
+from skimage import filters, io
+from skimage.morphology import disk
+
+from mpl_config import MPL_Config
+import gdal_virtual_file_system as gdal_vfs
+
+
+
+def cal_water_mask(row: Dict[str, Any], config: MPL_Config) -> Dict[str, Any]:
+    """
+    This will calculate the water mask to avoid (inference) processing of the masked areas with water
+    Uses gdal to transform the image into the required format.
+    The result of this function is that it will add a "mask" column with the water mask to each row.
+    It also writes the water mask to the config.WATER_MASK_DIR.
+    Parameters
+    ----------
+    row : Row in Ray Dataset, there is one row per image.
+    config : Contains static configuration information regarding the workflow.
+    """
+    worker_water_root = config.WATER_MASK_DIR
+    temp_water_root = (
+        config.TEMP_W_IMG_DIR
+    ) 
+
+    image_name = row["image_name"]
+    worker_water_subroot = os.path.join(worker_water_root, image_name)
+    temp_water_subroot = os.path.join(temp_water_root, image_name)
+    # Prepare to make directories to create the files
+    try:
+        shutil.rmtree(worker_water_subroot)
+    except:
+        print("directory deletion failed")
+        pass
+
+    try:
+        shutil.rmtree(temp_water_subroot)
+    except:
+        print("directory deletion failed")
+        pass
+
+    # check local storage for temporary storage
+    Path(worker_water_subroot).mkdir(parents=True, exist_ok=True)
+    Path(temp_water_subroot).mkdir(parents=True, exist_ok=True)
```

Comment:
if we are creating the directories using "exist_ok=true" then we should be able to remove the directory deletion logic above. Or is the deletion there to ensure that we don't have any artifacts present from previous runs?

Reply:
If I remember I think that was the reason - things left over from previous runs.

Reply:
Instead of doing it this way, I think it makes more sense to keep data from previous runs but have them dated so you know which data came from which runs. That might be helpful to know how the data produced from the different runs is changing over time. I'm not going to address that in this PR since it's out of scope but I'd recommend it for future work